### PR TITLE
Added support for LDAP Server Definitions

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.

### CODE REVIEW

File: docs/source/modules/zhmc_ldap_server_definition.rst [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+
+hmc_auth
+  The authentication credentials for the HMC.
+
+  | **required**: True
+  | **type**: dict
+
+
+  userid
```
Comment (EdGue42) [inline docs/source/modules/zhmc_ldap_server_definition.rst:49]: Indented by two spaces, does that have a purpose?

File: docs/source/modules/zhmc_ldap_server_definition_list.rst [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+- The HMC userid must have object-access permission to the target LDAP Server Definitions, or task permission to the 'Manage Users' task.
+
+
+
+
+Parameters
+----------
+
+
+hmc_host
```
Comment (EdGue42) [inline docs/source/modules/zhmc_ldap_server_definition_list.rst:34]: no "including" of duplicated content in rst?

File: plugins/modules/zhmc_ldap_server_definition.py [hunk 3]
```diff
@@ -0,0 +1,687 @@
...
+          zhmcclient.LDAPServerDefinition.update_properties()
+
+    Raises:
+      ParameterError: An issue with the module parameters.
+    """
+    create_props = {}
+    update_props = {}
+
+    # handle 'name' property
+    lsd_name = to_unicode(params['name'])
```
Comment (EdGue42) [inline plugins/modules/zhmc_ldap_server_definition.py:359]: so name is always present? (just wondering why this couldnt give a key error?)

File: plugins/modules/zhmc_ldap_server_definition.py [hunk 4]
```diff
@@ -0,0 +1,687 @@
...
+
+    log_file = module.params['log_file']
+    log_init(LOGGER_NAME, log_file)
+
+    _params = dict(module.params)
+    del _params['hmc_auth']
+    LOGGER.debug("Module entry: params: %r", _params)
+
+    try:
+
```
Comment (EdGue42) [inline plugins/modules/zhmc_ldap_server_definition.py:666]: empty line on purpose?

File: docs/source/modules/zhmc_ldap_server_definition.rst (hunk 1 above)
Comment (andy-maier) [reply docs/source/modules/zhmc_ldap_server_definition.rst:49]: This is a file generated by the docs tooling, and I don't know how to influence the layout of the generated RST (i.e. I'm not aware that we can configure a template or so).

File: docs/source/modules/zhmc_ldap_server_definition_list.rst (hunk 2 above)
Comment (andy-maier) [reply docs/source/modules/zhmc_ldap_server_definition_list.rst:34]: again, this is a generated file.

File: docs/source/modules/zhmc_ldap_server_definition_list.rst [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+   ---
+   # Note: The following examples assume that some variables named 'my_*' are set.
+
+   - name: List LDAP Server Definitions
+     zhmc_ldap_server_definition_list:
+       hmc_host: "{{ my_hmc_host }}"
+       hmc_auth: "{{ my_hmc_auth }}"
+     register: lsd_list
+
+
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply docs/source/modules/zhmc_ldap_server_definition_list.rst:109]: This is a generated file.

File: plugins/modules/zhmc_ldap_server_definition.py [hunk 6]
```diff
@@ -0,0 +1,687 @@
...
+#   eq_func: Equality test function for two values of the property; None means
+#     to use Python equality.
+#   type_cast: Type cast function for an input value of the property; None
+#     means to use it directly. This can be used for example to convert
+#     integers provided as strings by Ansible back into integers (that is a
+#     current deficiency of Ansible).
+ZHMC_LSD_PROPERTIES = {
+
+    # create-only properties:
+    'name': (False, True, False, None, casefold, None),
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply plugins/modules/zhmc_ldap_server_definition.py:289]: We could do something about that, such as using the built-in "dataclasses" or "attrs" package. mpython has a good Youtube video about using them. But that should be a separate PR across all modules. Also, the "dataclasses" package has been added only in Python 3.7, but maybe there is a backport (did not check). If you are keen to improving this, create an issue.

File: plugins/modules/zhmc_ldap_server_definition.py (hunk 3 above)
Comment (andy-maier) [reply plugins/modules/zhmc_ldap_server_definition.py:359]: Yes. For required parms and for optional parms that have a default defined, Ansible ensures that the parm is always present. That's why we now also simulate that behavior in our test invocation of the modules (see PR #740).

File: plugins/modules/zhmc_ldap_server_definition.py (hunk 4 above)
Comment (andy-maier) [reply plugins/modules/zhmc_ldap_server_definition.py:666]: Sometimes when I want to emphasize some line, I put blank lines around it in order to catch my attention easier when debugging. The following line is the main function that is called. I guess it is a personal style question.

It is possible to standardize such things by using the "black" tool. I don't like all things it does, but you may want to try it out and see what you think about it.

**Comment by andy-maier** [comment]
@EdGue42 Thanks for the review. I have responded to all comments. If you agree, please mark them as resolved and approve the PR.

**Review by EdGue42 — APPROVED** [review]

**Pull request merged**